django/db/backends/mysql/features.py

```diff
@@ -116,6 +114,19 @@ def needs_explain_extended(self):
         # EXTENDED is deprecated (and not required) in MySQL 5.7.
         return not self.connection.mysql_is_mariadb and self.connection.mysql_version < (5, 7)
 
+    @cached_property
+    def needs_analyze_statement(self):
+        # EXPLAIN ANALYZE was replaced by ANALYZE in MariaDB 10.1.0.
+        return self.connection.mysql_is_mariadb
```

Comment:
This database feature is confusing to me, I would prefer to use `mysql_is_mariadb` in `explain_query_prefix()`, e.g.
```python
def explain_query_prefix(self, format=None, **options):
    # Alias MySQL's TRADITIONAL to TEXT for consistency with other backends.
    # If TREE is supported, use that by default as it is more informative.
    if format and format.upper() == 'TEXT':
        format = 'TRADITIONAL'
    elif not format and 'TREE' in self.connection.features.supported_explain_formats:
        format = 'TREE'
    analyze = bool(options.pop('analyze', False))
    prefix = super().explain_query_prefix(format, **options)
    if analyze and self.connection.mysql_is_mariadb:
        # MariaDB uses ANALYZE instead of EXPLAIN ANALYZE.
        prefix = 'ANALYZE'
    ....
```

Reply:
That's fair - I can drop the feature. I added it as I thought there was precedence with `connection.features.needs_explain_extended`, but that will get removed at some point...